engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -870,6 +873,21 @@ private void execute() throws YarnException, IOException
       }
 
       numRequestedContainers += containerRequests.size() - removedContainerRequests.size();
+
```

Comment:
It might be better to populate pendingContainersToKill inside sendContainerAskToRM() where the last loop goes thru all of dnmgr.containerStopRequests and optionally issues a stopContainerAsync. At that point we can add the current containerIdStr  entry to pendingContainersToKill with the current-time. The logic is in one place so easier to understand (also possibly synchronized on containerStopRequests to prevent race conditions).